packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -268,9 +268,22 @@ module.exports = (
       if (cachedHeadings) {
         return cachedHeadings
       } else {
+        const getHeadingID = heading => {
+          const data = heading.data
+          if (data) {
+            if (data.id) return data.id
+            if (data.htmlAttributes && data.htmlAttributes.id) {
+              return data.htmlAttributes.id
+            }
```

Comment:
Can we also try to look into `data.hProperties`? The `gatsby-remark-autolink-headers` does add `id` in multiple ways (might be needed depending how it's consumed, not sure here):

https://github.com/gatsbyjs/gatsby/blob/87c54a44e795b379fad617c455aa53feb81e82e1/packages/gatsby-remark-autolink-headers/src/index.js#L61-L62